base/abstractdict.jl

```diff
@@ -335,6 +335,10 @@ function _typeddict(d::AbstractDict, others::AbstractDict...)
     Dict{K,V}(d)
 end
 
+reduce(m::typeof(merge), items::AbstractVector) = _merge!(mapreduce(typeof, typejoin, items), items)
+_toany(T) = isconcretetype(T) ? T : Any
+_merge!(::Type, items::AbstractVector) = reduce(merge, first(items), view(items,firstindex(items)+1:lastindex(items)))
```

Comment:
Break line before `reduce` to keep below 92 chars limit.

Reply:
> Break line
Ok.

> the second argument to reduce is only used for empty collections

Yes, that's usually correct. I don't recall if I innovated here or was copying something. But, it may be that `invoke` is more robust in any case.